### PR TITLE
Interoperability between Quarkus 3 and Hibernate ORM 6.x

### PR DESCRIPTION
**Pull request by ahus1**
This PR continues the work started in #17664

It includes the changes to make Hibernate 6 which was introduces in Quarkus 3.0.0.Alpha6 work with Keycloak.

### CODE REVIEW

**Review by sguilhen — APPROVED** [review]
Great work @mabartos  and @ahus1 . I agree that the failing tests don't seem related to Hibernate 6 upgrade.

Also, as every workaround is being tracked separately - which allows us to remove them once a fix is available in Hibernate - I'm approving this PR.

**Review by vmuzikar — COMMENTED** [review]
Thanks both @mabartos and @ahus1, nice work! Changes LGTM, added some questions I'd like to resolve before merge.

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java [hunk 1]
```diff
@@ -27,7 +27,6 @@ public static PropertyMapper[] getDatabasePropertyMappers() {
         return new PropertyMapper[] {
                 fromOption(DatabaseOptions.DB_DIALECT)
                         .mapFrom("db")
-                        .to("quarkus.hibernate-orm.dialect")
```
[the remark this answers is not in the document]
Comment (vmuzikar) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java:30]: @ahus1 Thanks for the clarification! Makes sense to look into this as a follow-up. Could you please create a GH Issue so we don't forget with all the details you have around it?

**Review by vmuzikar — APPROVED** [review]

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ahus1) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java:30]: Issue created: #19427 and assigned to the storage team. Still, we would need some help from the CND team AFAIK. cc: @hmlnarik

[the remark this answers is not in the document]
Comment (vmuzikar) [reply quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java:30]: Thanks and sure, let me know where can we help with #19427

**Pull request merged**